src/main/java/io/github/tetherless_world/jena_elasticsearch/ElasticsearchGraphMaker.java

```
@@ -0,0 +1,115 @@
+        BufferedReader br = new BufferedReader(new InputStreamReader(inputStream));
+        String line;
+        while ((line = br.readLine()) != null) {
+            resultStringBuilder.append(line).append("\n");
+        }
+        elasticsearchIndexSettings = resultStringBuilder.toString();
+
+        // Construct the high-level ES REST client
+        client = new RestHighLevelClient(RestClient.builder(httpHosts));
+
+        // Get all the graphs currently on this Elasticsearch cluster
+        // and add them to the locally tracked set
+        GetIndexRequest request = new GetIndexRequest("_all");
+        GetIndexResponse response = client.indices().get(request, RequestOptions.DEFAULT);
+        graphNames.addAll(Arrays.asList(response.getIndices()));
+    }
+
+    /**
+     * Return an ElasticsearchGraph with the given name
+     *
+     * @param name   the name of the ElasticsearchGraph
+     * @param strict if true, throw AlreadyExistsException a graph if a graph with the given name already exists
+     * @return
+     */
+    public Graph createGraph(String name, boolean strict) throws AlreadyExistsException {
+        // Elasticsearch only allows lowercase index names
+        String lowerName = name.toLowerCase();
+
+        if (graphNames.contains(lowerName)) {
+            // there is already a graph with this name
+            if (strict) {
+                // return the associated graph
+            } else {
+                throw new AlreadyExistsException("Graph '" + lowerName + "' already exists");
+            }
+        } else {
+            // there is no graph with this name yet
+            CreateIndexRequest request = new CreateIndexRequest(lowerName);
+            request.source(elasticsearchIndexSettings, XContentType.JSON);
+            try {
+                client.indices().create(request, RequestOptions.DEFAULT);
+            } catch (IOException e) {
+                e.printStackTrace();
```

Use SLF4J before you get into this habit..

---

In this case you'd want to bubble up the exception anyway. Wrap it in a RuntimeException maybe.